Automattic/wp-calypso
# A4A: Fix scan upsell link to make sure it goes to marketplace/products, plus header button style fixes

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/228

Also related to https://github.com/Automattic/automattic-for-agencies-dev/issues/233

## Proposed Changes

* This PR fixes the 'Add Jetpack Scan' button from the sites overview preview pane scan tab, to go to the marketplace and add a scan product / open the cart.
* It also fixes the link from the 'unassigned license' notice that can display on the same tab.
* Lastly, this PR modifies the styles of the A4A header action buttons to match what they were a day or so ago, before a recent PR changed the styles unintentionally.

## Testing Instructions

For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.

* Open the preview pane for any site and visit the Scan panel. If there is an upsell available, the 'Add Jetpack Scan' button should take you to the marketplace/products link, add the product to the cart, and open the cart popover.
* Also, if you have unassigned licenses available, a notice will display with a link to your unassigned licenses. That link now also goes to the unassigned licenses page in A4A.
* Lastly, the buttons at the top of the sites dashboard as well as the overview page now match the styles that existed a day or so ago.



Header button style recently looked like this after a PR changed the styles:
<img width="393" alt="Screenshot 2024-04-05 at 16 14 04" src="https://github.com/Automattic/wp-calypso/assets/16754605/b649c60f-db4c-456e-a005-d0c28f101b6a">


Header button style should look like this:
<img width="399" alt="Screenshot 2024-04-05 at 16 13 41" src="https://github.com/Automattic/wp-calypso/assets/16754605/b4710bb6-70bf-4031-8b96-9ca0340de673">

## Pre-merge Checklist



- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)https://github.com/Automattic/automattic-for-agencies-dev/issues/228
